libraries/opensk/src/env/test/rust_crypto.rs

```diff
@@ -0,0 +1,254 @@
+use core::convert::TryFrom;
+use p256::ecdh::EphemeralSecret;
+use p256::ecdsa::signature::{SignatureEncoding, Signer, Verifier};
+use p256::ecdsa::{SigningKey, VerifyingKey};
+use p256::elliptic_curve::sec1::ToEncodedPoint;
+use rand_core::OsRng;
+use rng256::Rng256;
+
+pub struct TestCrypto;
+pub struct TestEcdh;
+pub struct TestEcdsa;
+
+impl Crypto for TestCrypto {
+    type Ecdh = TestEcdh;
+    type Ecdsa = TestEcdsa;
+}
+
+impl ecdh::Ecdh for TestEcdh {
+    type SecretKey = TestEcdhSecretKey;
+    type PublicKey = TestEcdhPublicKey;
+    type SharedSecret = TestEcdhSharedSecret;
+}
+
+pub struct TestEcdhSecretKey {
+    ephemeral_secret: EphemeralSecret,
+}
+
+impl ecdh::SecretKey for TestEcdhSecretKey {
+    type PublicKey = TestEcdhPublicKey;
+    type SharedSecret = TestEcdhSharedSecret;
+
+    fn random(_rng: &mut impl Rng256) -> Self {
+        let ephemeral_secret = EphemeralSecret::random(&mut OsRng);
+        Self { ephemeral_secret }
+    }
+
+    fn public_key(&self) -> Self::PublicKey {
+        let public_key = self.ephemeral_secret.public_key();
+        TestEcdhPublicKey { public_key }
+    }
+
+    fn diffie_hellman(&self, public_key: &TestEcdhPublicKey) -> Self::SharedSecret {
+        let shared_secret = self.ephemeral_secret.diffie_hellman(&public_key.public_key);
+        TestEcdhSharedSecret { shared_secret }
+    }
+}
+
+pub struct TestEcdhPublicKey {
+    public_key: p256::PublicKey,
+}
+
+impl ecdh::PublicKey for TestEcdhPublicKey {
+    fn from_coordinates(
+        x: &[u8; EC_FIELD_BYTE_SIZE],
+        y: &[u8; EC_FIELD_BYTE_SIZE],
+    ) -> Option<Self> {
+        let encoded_point: p256::EncodedPoint =
+            p256::EncodedPoint::from_affine_coordinates(x.into(), y.into(), false);
+        let public_key = p256::PublicKey::from_sec1_bytes(encoded_point.as_bytes()).ok()?;
+        Some(Self { public_key })
+    }
+
+    fn to_coordinates(&self, x: &mut [u8; EC_FIELD_BYTE_SIZE], y: &mut [u8; EC_FIELD_BYTE_SIZE]) {
+        let point = self.public_key.to_encoded_point(false);
+        x.copy_from_slice(point.x().unwrap());
+        y.copy_from_slice(point.y().unwrap());
+    }
+}
+
+pub struct TestEcdhSharedSecret {
+    shared_secret: p256::ecdh::SharedSecret,
+}
+
+impl ecdh::SharedSecret for TestEcdhSharedSecret {
+    fn raw_secret_bytes(&self) -> [u8; EC_FIELD_BYTE_SIZE] {
+        let mut bytes = [0; EC_FIELD_BYTE_SIZE];
+        bytes.copy_from_slice(self.shared_secret.raw_secret_bytes().as_slice());
+        bytes
+    }
+}
+
+impl ecdsa::Ecdsa for TestEcdsa {
+    type SecretKey = TestEcdsaSecretKey;
+    type PublicKey = TestEcdsaPublicKey;
+    type Signature = TestEcdsaSignature;
+}
+
+pub struct TestEcdsaSecretKey {
+    signing_key: SigningKey,
+}
+
+impl ecdsa::SecretKey for TestEcdsaSecretKey {
+    type PublicKey = TestEcdsaPublicKey;
+    type Signature = TestEcdsaSignature;
+
+    fn random(_rng: &mut impl Rng256) -> Self {
+        let signing_key = SigningKey::random(&mut OsRng);
```

Comment:
Shouldn't we implement `CryptoRngCore` for `Rng256` when `rust_crypto` is set and use the `rng` argument instead of `OsRng` (which is counter-intuitive)? Same for the other usages of `OsRng` in this file.

Reply:
I will, eventually! This is a 2 line fix before I work on RNG.

Reply:
Ok I see. Let's add a TODO comment, create an issue, or already send the next PR then to not forget.

Reply:
Done